src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -671,13 +671,30 @@ export function getAppointmentSlots(startDate, endDate, forceFetch = false) {
   };
 }
 
-export function onCalendarChange(selectedDates) {
+export function onCalendarChange(
+  selectedDates,
+  maxSelections,
+  upcomingAppointments,
+) {
+  let isSame = false;
+  if (maxSelections === 1 && selectedDates.length > 0) {
+    const date = selectedDates[0];
+
+    const d1 = moment(date, 'YYYY-MM-DDTHH:mm:ss');
+    const appointments = upcomingAppointments[d1.format('YYYY-MM')];
+
+    isSame = appointments?.some(appointment => {
```

Comment:
Hmm this wasn't what I was expecting so I'm curious if we thought of working with time interval overlaps. Specifically, we'll create intervals using `appointment.start` and `appointment.minutesDuration`. Then we can compare interval overlaps using [areIntervalsOverlapping](https://date-fns.org/v4.1.0/docs/areIntervalsOverlapping)

Reply:
I did ask Ciera about this and she did not remember any such conversation. This could be addressed in a new ticket though.

Reply:
Sounds good!

Reply:
Also, 'moment.isSame' is timezone aware. https://momentjscom.readthedocs.io/en/latest/moment/05-query/02-is-same/

Reply:
`upcomingAppointments` contains both booked and cancelled future appointments. You'll want to exclude cancelled appointments so it does get compared against the selected date creating a false positive to show the duplicate error.

Reply:
Thanks Jenny!

Reply:
@brianseek did some work on this a while back. I don't think this takes into account cancelled appointments as @jenniemc mentioned but it gets at the intervals that you mentioned @JunTaoLuo. https://github.com/department-of-veterans-affairs/vets-website/blob/d39b983b48202e96a2b4965726fb1b47b96d8d21/src/applications/vaos/referral-appointments/utils/provider.js#L604

@vbahinwillit we could probably use it or something like it in that future ticket.